twenty-first/src/tip5/mod.rs

```diff
@@ -1067,6 +1089,50 @@ pub(crate) mod tests {
         assert_eq!(Tip5::offset_fermat_cube_map(255), 255);
     }
 
+    /// Ensure that the claims made in [`Tip5::mds_generated`] are true.
+    ///
+    /// In particular, `BFieldElement::Add` internally uses the equality
+    /// `a + b = a - (p - b)`. If `a` is a degenerate representation (i.e., the
+    /// Montgomery representation is larger than or equal to
+    /// [`BFieldElement::P`]), then a small enough `b` makes the result of the
+    /// addition non-degenerate by removing the “surplus” from `a`. In
+    /// particular, this correction will happen if the following inequality
+    /// holds:
+    ///              p - b > u64::MAX - p
+    ///   ↔          p - b > u32::MAX
+    ///   ↔   p - u32::MAX > b
```

Comment:
I get a much tighter bound: `p - b > u32::MAX - 1`. Because expanding `u64::MAX - p = 2^64 - 1 - 2^64 + 2^32 - 1 = 2^32 - 1 - 1`.

Reply:
I.e., `p - u32::MAX + 1 > b`? That’s indeed much tighter. 😁 Yes, seems like I’m off by one.